pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3922,4 +3922,66 @@ protected void internalHandleResult(AsyncResponse asyncResponse,
             }
         }
     }
+
+    protected void internalTruncateTopic(AsyncResponse asyncResponse) {
+
```

Comment:
We should check the topic ownership first and then get the topic reference. So that we can use the ManagedLedger of the topic to truncate directly.

Reply:
+1

Reply:
I see. Thank you very much. I will modify them in the way of check the topic ownership first and then get the topic reference. At the same time, The implementation is moved from brokerService to Topic